### ISSUE TITLE
The mapper create a child's instance instead of a parent's instance.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#34]
```
What steps will reproduce the problem?
1. Make inheritance child from parent.
2. Get a child bean from the MapperFacade.
3. Get a parent bean from the MapperFacade

What is the expected output? What do you see instead?
Parent's instance is expected. I see child's instance instead.

What version of the product are you using? On what operating system?
Orika 1.1.8

Please provide any additional information below.
see attached test
```

Original issue reported on code.google.com by `dkhomya...@gmail.com` on 20 Jul 2012 at 6:43

Attachments:
- [ParentChildTest.java](https://storage.googleapis.com/google-code-attachments/orika/issue-34/comment-0/ParentChildTest.java)

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.deb...@gmail.com` on 4 Aug 2012 at 9:14
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This seems to happen because Orika favors the use of an existing mapper (which 
satisfies the inputs) over generating a new one.

So since the Entity<=>Child mapping happened first (and created a mapping), 
this one is found when Entity<=>Parent is passed in.

Perhaps Orika should not consider auto-generated mappings when making this 
decision...

Anyway, in the 1.2.0-branch, I find that the desired behavior can be achieved 
by adding an explicit mapping for Entity<=>Parent. 
```

Original comment by `matt.deb...@gmail.com` on 4 Aug 2012 at 4:36

**Comment by GoogleCodeExporter** [comment]
```
Unfortunally, an explicit mapping isn't suitable for me.
```

Original comment by `dkhomya...@gmail.com` on 8 Aug 2012 at 8:15

**Comment by GoogleCodeExporter** [comment]
```
A fix has been applied in the 1.2.0 branch; please check and confirm if you get 
a chance...
```

Original comment by `matt.deb...@gmail.com` on 17 Aug 2012 at 10:53
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
I've tested 1.2.0-branch. All right. Thanks.
```

Original comment by `dkhomya...@gmail.com` on 20 Aug 2012 at 1:01
